scripts/azureml-assets/azureml/assets/config.py

```diff
@@ -345,7 +351,73 @@ def __init__(self, branch: str, uri: str):
         self._branch = branch
         super().__init__(PathType.GIT, uri)
 
+class DataConfig(Config):
+    """Data Config class."""
+
+    """
+    Example:
+
+    path: #Could be azure storage, git , hugging face
+
+        ## Azure Blobstore example
+        type: azureblob
+        storage_name: my_storage
+        container_name: my_container
+        container_path: foo/bar
+    """
+    def __init__(self, file_name: Path):
+        """Initialize object for the Data Properties extracted from extra_config data.yaml."""
+        super().__init__(file_name)
+        self._path = None
+        self._validate()
+
+    def _validate(self):
+        """Validate the yaml file."""
+        Config._validate_exists('data.path', self.path)
+        Config._validate_enum('data.path.type', self.path.type.value, PathType, True)
+        Config._validate_enum('data.type', self._type, DataType, True)

```

Comment:
These should operate on raw values, not `self.path`. Please create a `_path` property that returns the raw `self._yaml.get('path')` value and work with that. You'll of course need to use another private variable to store the `AssetPath` generated by `path`.